rules-reviewed/jboss-eap/weblogic/weblogic-services.windup.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="weblogic-services">
+    <metadata>
+        <description>
+            This ruleset contains rules to detect WebLogic specific services code (eg, startup listeners, and access to WebLogic specific service
+            lookup utilities), and to assist in the conversion of this code to code that is compatible with the JBoss Enterprise Application Platform.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,2.3.0.Final"/>
+            <addon id="org.jboss.windup.rules,windup-rules-java,2.3.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="weblogic" versionRange="(1,)"/>
+        <targetTechnology id="eap" versionRange="[6,)"/>
+        <tags>
+            <tag>reviewed-2015-05-21</tag>
+        </tags>
+    </metadata>
+    <rules>
+        <rule id="weblogic-services-01000">
+            <when>
+                <javaclass references="weblogic.common.T3StartupDef">
+                    <location>IMPLEMENTS_TYPE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <classification title="WebLogic Startup Service" effort="3" severity="mandatory">
+                    <link href="http://docs.oracle.com/javaee/6/api/javax/ejb/Singleton.html" description="EJB3.1 Singleton Bean"/>
+                    <link href="http://docs.oracle.com/javaee/6/api/javax/ejb/Startup.html" description="EJB3.1 Startup Bean"/>
+                </classification>
+                <hint title="Use of WebLogic Startup Service" effort="0" severity="mandatory">
+                    <message>Replace this class with a class using the EJB 3.1 @Singleton and @Startup annotations.</message>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="weblogic-services-02000">
+            <when>
+                <javaclass references="weblogic.common.T3ServicesDef"/>
+            </when>
+            <perform>
+                <hint title="Use of T3ServicesDef" effort="3" severity="mandatory">
+                    <message>
+                        T3ServicesDef provides access to core services of the container, such as Timers and Logging facilities.
+
+                        Replace the services provided by this with a @Singleton EJB that provides access to
```

Comment:
Is JBoss a product name? this should probably be rephrased:

"that provides access to the equivalent services from JBoss Enterprise Application Platform"